packages/ui/src/components/notifications/NotificationCenter.test.tsx

```diff
@@ -0,0 +1,169 @@
+
+describe('NotificationCenter', () => {
+    const mockNotification = (message: string) => {
+        const notification: Partial<Notification> = {
+            type: 'info',
+            message,
+            createdAt: new Date(),
+            id: randomString(),
+        };
+        return notification;
+    };
+    const emitEvent = (message: string) => {
+        const event: CustomEvent<Partial<Notification>> = new CustomEvent(NOTIFICATION_EVENT, {
+            detail: mockNotification(message),
+        });
+        fireEvent(document, event);
+    };
+    const SnackComponent = ({ onClose, id, message }: SnackbarProps) => (
+        <button type="button" onClick={() => onClose(id)} key={id}>
+            {message}
+        </button>
+    );
+    const renderComponent = (props: { hideAfter?: number; maxNotifications?: number; cacheTimeout?: number } = {}) => {
+        return render(<NotificationCenter SnackbarComponent={SnackComponent} {...props} />);
+    };
+
+    beforeEach(() => {
+        vi.useFakeTimers();
+    });
+
+    it('should render a notification', () => {
+        const { getByText } = renderComponent();
+
+        emitEvent('test');
+
+        expect(getByText('test')).toBeInTheDocument();
+    });
+
+    it("should render a stack of notifications, up to the component's maxNotifications prop", () => {
+        const { queryByText } = renderComponent();
+
+        waitFor(() => {
+            emitEvent('test-1');
+            emitEvent('test-2');
+            emitEvent('test-3');
+            emitEvent('test-4');
+            emitEvent('test-5');
+            emitEvent('test-6');
+        });
+
+        expect(queryByText('test-1')).not.toBeInTheDocument();
+        expect(queryByText('test-2')).toBeInTheDocument();
+        expect(queryByText('test-3')).toBeInTheDocument();
+        expect(queryByText('test-4')).toBeInTheDocument();
+        expect(queryByText('test-5')).toBeInTheDocument();
+        expect(queryByText('test-6')).toBeInTheDocument();
+    });
+
+    it('should hide notifications after the hideAfter prop', () => {
+        const { queryByText } = renderComponent({ hideAfter: 1000 });
+
+        emitEvent('test-1');
+
+        expect(queryByText('test-1')).toBeInTheDocument();
+
+        act(() => {
+            vi.advanceTimersByTime(1000);
+        });
+
+        waitFor(() => {
+            expect(queryByText('test-1')).not.toBeInTheDocument();
+        });
+    });
+
+    it('should hide notifications after the cacheTimeout prop', () => {
+        const { queryByText, getByTestId } = renderComponent({ cacheTimeout: 1000 });
+
+        emitEvent('test-1');
+        emitEvent('test-2');
+        emitEvent('test-3');
+
+        expect(queryByText('test-1')).toBeInTheDocument();
+        expect(queryByText('test-2')).toBeInTheDocument();
+        expect(queryByText('test-3')).toBeInTheDocument();
+
+        act(() => {
+            vi.advanceTimersByTime(1000);
+        });
+
+        waitFor(() => {
```

Comment:
the test will always pass if you don't await this, since it's a promise and it won't get called before the test finishes with success. Same for the other tests